packages/js/src/generators/library/library.ts

```diff
@@ -46,22 +46,20 @@ export async function libraryGenerator(
   tree: Tree,
   schema: LibraryGeneratorSchema
 ) {
-  const { layoutDirectory, projectDirectory } = extractLayoutDirectory(
-    schema.directory
-  );
-  schema.directory = projectDirectory;
-  const libsDir = schema.rootProject
-    ? '.'
-    : layoutDirectory ?? getWorkspaceLayout(tree).libsDir;
-  return projectGenerator(tree, schema, libsDir, join(__dirname, './files'));
+  return await libraryGeneratorInternal(tree, {
+    // provide a default nameDirectoryFormat to avoid breaking changes
+    // to external generators invoking this one
+    nameDirectoryFormat: 'derived',
+    ...schema,
+  });
 }
 
-export async function projectGenerator(
+export async function libraryGeneratorInternal(
   tree: Tree,
-  schema: LibraryGeneratorSchema,
-  destinationDir: string,
-  filesDir: string
```

Comment:
Were these used somewhere?

Reply:
Yes, but they are no longer needed with the changes introduced in the PR.